functional/unit_action_test.go

```diff
@@ -224,3 +229,151 @@ func TestUnitSSHActions(t *testing.T) {
 		t.Errorf("Could not find expected string in journal output:\n%s", stdout)
 	}
 }
+
+// TestUnitDestroyFromRegistry() checks for a submitted unit being removed
+// from the etcd registry. It compares a local unit body with the unit in
+// the etcd registry, to verify the body is identical.
+func TestUnitDestroyFromRegistry(t *testing.T) {
+	cluster, err := platform.NewNspawnCluster("smoke")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cluster.Destroy()
+
+	m, err := cluster.CreateMember()
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = cluster.WaitForNMachines(m, 1)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// submit a unit and assert it shows up
+	if _, _, err := cluster.Fleetctl(m, "submit", "fixtures/units/hello.service"); err != nil {
+		t.Fatalf("Unable to submit fleet unit: %v", err)
+	}
+	var stdout string
+	stdout, _, err = cluster.Fleetctl(m, "list-units", "--no-legend")
+	if err != nil {
+		t.Fatalf("Failed to run list-units: %v", err)
+	}
+	units := strings.Split(strings.TrimSpace(stdout), "\n")
+	if len(units) != 1 {
+		t.Fatalf("Did not find 1 unit in cluster: \n%s", stdout)
+	}
+
+	// cat the unit and compare it with the value in etcd registry
+	if stdout, _, err = cluster.Fleetctl(m, "cat", "hello.service"); err != nil {
+		t.Fatalf("Unable to retrieve the fleet unit: %v", err)
+	}
+	unitBody := stdout
```

Comment:
Don't use a temporary `stdout` -- just assign the result to `unitBody` directly.

(Note: This will require the assignment to be moved out of the `if` -- but that's the common pattern used throughout the existing codebase...)